cmd/buddha.go

```diff
@@ -176,22 +178,37 @@ func runJob(job *buddha.Job) error {
 	for _, cmd := range job.Commands {
 		log.Println(log.LevelPrim, "Command: %s", cmd.Name)
 
+		log.Println(log.LevelScnd, "Executing necessity checks")
+		isNecessaryResults, err := executeChecks(cmd, cmd.Necessity, executeNecessityCheck)
+		if err != nil {
+			log.Println(log.LevelFail, "fatal: unexpected error from necessity check, ending run")
+			return err
+		}
+		if allFalse(isNecessaryResults) {
+			if *OnUnnecessary == "continue" {
+				log.Println(log.LevelFail, "warning: job unnecessary, continuing anyway")
+			} else {
+				log.Println(log.LevelInfo, "Job deemed unnecessary, skipping")
+				continue
+			}
+		}
+
 		// execute before health checks
 		// these will execute once and depending on --on-before-fail skip this job
 		log.Println(log.LevelScnd, "Executing before checks")
-		result, err := executeChecks(cmd, cmd.Before, 1)
+		checksResults, err := executeChecks(cmd, cmd.Before, executeHealthCheck)
 		if err != nil {
-			log.Println(log.LevelFail, "fatal: before checks failed, ending run")
+			log.Println(log.LevelFail, "fatal: unexpected error from before check, ending run")
 			return err
 		}
-		if !result {
+		if anyFalse(checksResults) {
 			if *OnBeforeFail == "stop" {
-				log.Println(log.LevelFail, "fatal: before checks returned false, ending run")
+				log.Println(log.LevelFail, "fatal: before returned false, ending run")
 				return nil
 			} else if *OnBeforeFail == "continue" {
-				log.Println(log.LevelFail, "warning: before checks returned false, continuing anyway")
+				log.Println(log.LevelFail, "warning: before returned false, continuing anyway")
 			} else {
-				log.Println(log.LevelInfo, "Before checks returned false, skipping job")
+				log.Println(log.LevelFail, "warning: before returned false, skipping job")
 				continue
 			}
```

Comment:
> Yeah overall i'm not too happy with the split in configuration between command line and config files. I feel we should consolidate at some point.

I'm not really sure which bit you're getting at - the `--on-*` command line flags are command line flags precisely because this is behaviour which is useful to change between runs.